dspace-api/src/main/java/org/dspace/app/util/IndexVersion.java

```diff
@@ -252,11 +252,8 @@ public static int compareSoftwareVersions(String firstVersion, String secondVers
             return GREATER_THAN;
         } else if (firstMinor < secondMinor) {
             return LESS_THAN;
-        } else {
-            // This is an impossible scenario.
-            // This 'else' should never be triggered since we've checked for equality above already
-            return EQUAL;
         }
+        return EQUAL;
```

Comment:
I think that this is also dead code:  at line 250 we know that firstMajor == secondMajor and firstMinor != secondMinor, so there is no possibility of reaching this line.

Reply:
yeah, so I think a suitable way to avoid the dead code is remove the `if (firstMinor < secondMinor)` and directly  return `LESS_THAN;` . The detail is as follow:

           `
          else if (firstMinor > secondMinor) { 

            // If we get here, major versions must be EQUAL. Now, time to check our minor versions

            return GREATER_THAN;

            } else {

            return LESS_THAN;

            }
`

Reply:
That is neater code, and easier to understand.  It builds cleanly for me.  There is a unit test for this method, which succeeds with the patch applied.